client/signup/controller.js

```diff
@@ -67,8 +68,8 @@ export default {
 	},
 
 	redirectToFlow( context, next ) {
-		if ( context.path !== utils.getValidPath( context.params ) ) {
-			return page.redirect( utils.getValidPath( context.params ) );
+		if ( context.pathname !== utils.getValidPath( context.params ) ) {
+			return page.redirect( addQueryArgs( context.query || {}, utils.getValidPath( context.params ) ) );
```

Comment:
Unless I'm mistaken, `context.query` is always an object, so you don't need `|| {}` here.

Also, this would obviate a lot of function calls:

```js
return page.redirect( utils.getValidPath( context.params ) + context.querystring ? '?' + context.querystring : '' );
```